File: ios/Runner/BackgroundHandler.swift
import Foundation

/// Keep-alive session control (ADR-005): iOS background audio session.
///
/// Phase 0 stub: real background audio session setup lands on-device.
final class BackgroundHandler: BackgroundApi {
  func startForegroundSession(completion: @escaping (Result<Void, Error>) -> Void) {
    completion(.failure(NotImplemented.error("BackgroundApi.startForegroundSession")))
  }

  func stopForegroundSession(completion: @escaping (Result<Void, Error>) -> Void) {
    completion(.failure(NotImplemented.error("BackgroundApi.stopForegroundSession")))
  }
}

/// Shared failure for every Phase 0 stub. Fails loudly (never a dummy value)
/// so Dart-side logic cannot mistake an unimplemented bridge for a working one.
enum NotImplemented {
  static func error(_ method: String) -> NSError {
    NSError(
      domain: "com.fzrilsh.toice.NotImplemented",
      code: 501,
      userInfo: [NSLocalizedDescriptionKey: "\(method) not implemented (Phase 0)"]
    )
  }
}

File: android/app/src/main/kotlin/com/fzrilsh/toice/BackgroundHandler.kt
package com.fzrilsh.toice

/**
 * Keep-alive session control (ADR-005): foreground service with a persistent
 * notification.
 *
 * Phase 0 stub: real foreground service lands on-device.
 */
class BackgroundHandler : BackgroundApi {
  override fun startForegroundSession(callback: (Result<Unit>) -> Unit) =
    callback(notImplemented("BackgroundApi.startForegroundSession"))

  override fun stopForegroundSession(callback: (Result<Unit>) -> Unit) =
    callback(notImplemented("BackgroundApi.stopForegroundSession"))
}

/**
 * Shared failure for every Phase 0 stub. Fails loudly (never a dummy value)
 * so Dart-side logic cannot mistake an unimplemented bridge for a working one.
 */
internal fun <T> notImplemented(method: String): Result<T> =
  Result.failure(NotImplementedError("$method not implemented (Phase 0)"))
